pulp_ansible/app/viewsets.py

```diff
@@ -88,6 +89,47 @@ class RoleViewSet(ContentViewSet):
     queryset = Role.objects.all()
     serializer_class = RoleSerializer
     filterset_class = RoleFilter
+    queryset_filtering_required_permission = "ansible.view_role"
+
+    DEFAULT_ACCESS_POLICY = {
+        "statements": [
+            {
+                "action": ["list", "retrieve"],
+                "principal": "authenticated",
+                "effect": "allow",
+                "condition": "has_repository_model_or_obj_perms:ansible.view_role",
```

Comment:
When `queryset_scoping` is specified, **Content** Viewsets will scope the returned results based on permissions the user has on repositories, so these two lines are not needed.

```suggestion

    DEFAULT_ACCESS_POLICY = {
        "statements": [
            {
                "action": ["list", "retrieve"],
                "principal": "authenticated",
                "effect": "allow",
```